src/Conf.ml

```diff
@@ -606,4 +621,22 @@ let action =
 
 and debug = !debug
 
+let ending_type file =
+  match config.ending_type with
+  | `Preserve -> (
+      let chan = Stdio.In_channel.create ~binary:false file in
+      let line = Stdio.In_channel.input_line ~fix_win_eol:false chan in
+      Stdio.In_channel.close chan ;
+      match line with
+      | Some line when String.is_substring line ~substring:"\r\n" ->
+          `Windows
+      | _ -> `Unix )
+  | `Unix -> `Unix
+  | `Windows -> `Windows
+
+let ending_token file =
+  match ending_type file with `Unix -> "\n" | `Windows -> "\r\n"
+
+let current_ending_token = ref "\n"
```

Comment:
I don't see why this can't be removed, it seems to be only written but never read?

Reply:
I intended to use it in place hard-coded '\n' in Fmt_ast, don't know if it is necessary at all (need some tests).